src/Carousel.js

```diff
@@ -0,0 +1,149 @@
+import React, { useState, useEffect, useRef } from 'react';
+import './Carousel.css';
+
+function Carousel() {
+  let SliderArr = ['Первый', 'Второй', 'Последний'];
+
+  const [x, setX] = useState(0);
+  const [startX, setStartX] = useState(0);
+  const [offsetX, setOffsetX] = useState(0);
+
+  const [mouseDown, setMouseDown] = useState(false);
+
+  const widthOfTheSlide = 100;
+  const standardOfDifference = 50;
+
+  // Вот эта херня нужна чтобы включать transition только при смене слайда
+  const [timer, setTimer] = useState(0);
+  const [timerGo, setTimerGo] = useState(false);
+
+  const id = useRef(0);
+
+  const TIMER_TIME = 1; // время transition в секундах
+
+  const clear = () => {
+    window.clearInterval(id.current);
+  };
+
+  React.useEffect(() => {
+    id.current = window.setInterval(() => {
+      setTimer((time) => time - 1);
+    }, 1000);
+    return () => clear();
+  }, [timerGo]);
+
+  // чтобы таймер не ушел в минус
```

Comment:
Pls, use English for comment, bcs our code may live longer than our work in it and someone may not know Russian